ws-nextjs-app/cypress/support/commands.ts

```diff
@@ -17,9 +17,45 @@ declare global {
         service: Services;
         pageType: PageTypes;
         variant?: Variants;
-        id: string;
+        id?: string;
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
       }) => Chainable<any>;
     }
   }
 }
+
+const getPageDataFromWindow = ({
+  service,
+  pageType,
+  variant,
+  id,
+}: {
+  service: Services;
+  pageType: PageTypes;
+  variant?: Variants;
+  id?: string;
+}) => {
+  const baseUrl =
+    Cypress.env('APP_ENV') === 'local'
+      ? 'http://localhost:7081'
+      : Cypress.config().baseUrl;
+  let path = `/${service}/${pageType}/${id}${variant ? `/${variant}` : ''}`;
+
+  if (service === 'ws') {
+    path = '/ws/languages';
+  }
+
+  cy.visit(`${baseUrl}${path}`);
```

Comment:
I don't think you need to do this - you are already on the page at this point - so we just need to get the data from the window.
```suggestion
```